File: InputTestCode/Swift/KotlinBasics/10_ControlFlow.swift
// Control flow

func forIn(items: [String]) -> String {
    var testString = "";

    for item in items {
        testString += item
    }

    return testString
}

func forInIndex(items: [String]) -> Int {
    var testValue = 0

    for (index, item) in items.enumerated() {
        testValue += index
    }

    return testValue
}

func forInIndexPlaceholder(items: [String]) -> Int {
    var testValue = 0

    for (index, _) in items.enumerated() {
        testValue += index
    }

    return testValue
}

func whileLoop() -> Int {
    var count = 0

    while (count < 15) {
        count += 1
    }

    return count
}

func describe(obj: Any) -> String {
    if(obj is Int && ((obj as? Int) == 1)) {
        return "One"
    } else if(obj is String && (obj as? String) == "Hello") {
        return "Greeting"
    } else if(obj is Int64) {
        return "Long"
    } else if(!(obj is String)) {
        return "Not a string"
    } else {
        return "Unkown"
    }
}

func translateNumber(num: Int) -> String {
    switch (num) {
        case 1:
            return "One"
        case 2:
            return "Two"
        case 3:
            return "Three"
        case 4:
            return "Four"
        case 5:
            return "Five"
        default:
            return "Number not known."
    }
}

File: InputTestCode/Kotlin/KotlinBasics/10_ControlFlow.kt
package original
// Control flow

fun forIn(items: Array<String>): String {
    var testString = "";

    for (item in items) {
        testString += item
    }

    return testString
}

fun forInIndex(items: Array<String>): Int {
    var testValue = 0

    for (index in items.indices) {
        testValue += index
    }

    return testValue
}

fun forInIndexPlaceholder(items: Array<String>): Int {
    var testValue = 0

    for (index in items.indices) {
        testValue += index
    }

    return testValue
}

fun whileLoop(): Int {
    var count = 0

    while (count < 15) {
        count += 1
    }

    return count
}

fun describe(obj: Any): String =
	when (obj) {
		1 -> "One"
		"Hello" -> "Greeting"
		is Long -> "Long"
		!is String -> "Not a string"
		else -> "Unknown"
	}

fun translateNumber(num: Int): String {
    when (num) {
        1 -> return "One"
        2 -> return "Two"
        3 -> return "Three"
        4 -> return "Four"
        5 -> return "Five"
        else -> return "Number not known."
    }
}